docs/ARCHITECTURE.md

```diff
@@ -9,10 +22,50 @@ can be configured to run only during certain time periods or not at all. Setting
 make sure repair is spread out over the interval time while alarms are provided to signal when a job has not run for
 longer than expected.
 
+<div align="center">
+
+  ```mermaid
+  flowchart RL
+      A(((Cassandra Node)))---B(((ecChronos Instance)));
+      A(((Cassandra Node)))---C(((ecChronos Instance)));
+      B(((ecChronos Instance)))---D(((Cassandra Node)));
+      C(((ecChronos Instance)))---E(((Cassandra Node)));
+      D(((Cassandra Node)))---F(((ecChronos Instance)));
+      E(((Cassandra Node)))---G(((ecChronos Instance)));
+      F(((ecChronos Instance)))---H(((Cassandra Node)));
+      G(((ecChronos Instance)))---H(((Cassandra Node)));
+  ```
+
+  <figcaption>Figure 1: ecChronos and Cassandra Nodes.</figcaption>
+</div>
+
 ## Concepts
 
 ### Leases
 
+In the context of Apache Cassandra, a "lease" refers to a mechanism employed during the repair process within Cassandra.
```

Comment:
This is not true, in context of Cassandra there's no "lease" during repair process. "Lease" is something you have to implement on the application level (like ecchronos, already described below)